satpy/etc/writers/awips_tiled.yaml

```diff
@@ -66,9 +66,12 @@ templates:
 #        value: "${ORGANIZATION}"
       awips_id: {}
 #        value: "{awips_id}"  # special variable created by awips_tiled.py
+      physical_element: {}
+#        value: "{physical_element}"   #special variable created by awips_tiled.py
```

Comment:
Why does this need to be handled by that fancy method? In the case of the polar template we only ever send one data array at a time to be rendered (if this isn't true then it is a bug) by the template class.

Oh...you need each variable to have a custom physical_element so that information needs to come from the variable element...interesting. I guess this makes sense then because the only other ways I could see this being done are:

1. Don't allow custom ones and create a clavrx template that makes the global physical element `"CLAVR-x {name}"`.
2. Create a separate template for each clavrx variable...very verbose.

Could you please verify that by adding that method in the python code that the GLM output doesn't have a global physical element? You could print out the attrs from one of the GLM test cases just to double check. Thanks.

Reply:
I thought about creating a separate template for each variable and thought that might be a pain in the long term.  It is true, that in the polar template only one variable is sent at a time (as far as I can tell).  You are correct, I need a custom physical_element for every variable element and that physical_element has to be global, AWIPS does not ingest the file without the global physical_element.  That must be unique because that is how AWIPS assigns the file path when ingesting (as I understand it).  

In the first solution, you are saying that instead of using the "polar" template, I create a separate clavrx template?  How will the awips_tiled writer know to use the clavrx template?  In this case, I remove the fancy method from the awips_tiled writer.

Reply:
AWIPS uses the physical element for the display name usually in the product browser. It is configured for "Polar" files here:

https://github.com/ssec/polar2grid/blob/f19c604d03c0f4fc1452a7ec48a1b484f2349cf1/awips_scmi/Polar/PolarCommonDescription.xml#L17-L19

>  I create a separate clavrx template? How will the awips_tiled writer know to use the clavrx template? In this case, I remove the fancy method from the awips_tiled writer.

Yes. You can provide the template name as a keyword argument to the writer. It is available in Polar2Grid here:

https://github.com/ssec/polar2grid/blob/f19c604d03c0f4fc1452a7ec48a1b484f2349cf1/polar2grid/writers/awips_tiled.py#L165-L170

We use it for the CSPP Gridded GLM project to select the GLM-specific templates. There could be one for CLAVR-x if that makes the most sense and is the easiest to maintain. However, I am not opposed to the solution you went with here. It is unfortunate, but I don't see a way around it while keeping the rest of the code the same.

If you keep it as you have it here could you provide a docstring to the method to explain what it is doing and why?

Reply:
I checked the glm and in the code does add a physical element to the global physical element, so I will have to try a different path.

Reply:
Correction, if I use template="glm_l2_radc" as a keyword in save_datasets, no global physical element in written.  However, the awips_tiled writer default polar template at least writes a file with glm data In that case, the global physical element is written.

Reply:
Ah ok, that is expected. Sounds good.

Reply:
The change to the awips_tiled.yaml and code to add the global physical attribute is okay then?

Reply:
Yes, I think so. I'm not sure there is a cleaner way to do it but that attribute is definitely needed for that template. So unless the CLAVR-x project is ok with generated `{name}` physical_element attributes then I'd say leave it as-is. It defaults to `{name}` right?

Reply:
I am no sure what you are asking when you write 
> unless the CLAVR-x project is ok with generated `{name}` physical_element attributes

Are you asking if the clavrx group would generate the physical element attribute in the original netcdf file?  I am not sure that would work well and then there would still need to be code which moves the physical element attribute from the variable to the global.

On the second question:
> It defaults to {name} right?

Yes.

Reply:
I meant the Polar template could be updated to have the global physical_element set to `{name}` which would mean the satpy product name would always be the physical_element in AWIPS. This would mean CLAVR-x files wouldn't have any "CLAVR-x {name}" prefix. If `{name}` was ok (granted it is not very flexible) then we wouldn't need the custom global attribute method you added.

Forget I said anything. Leave it as is.

Reply:
Let me check how that attribute is being used to ingest.  I have a feeling that product names will collide without the "CLAVR-x" descriptor before the product name.  However, AWIPS ingest is new to me and I need to check.

Reply:
I think we should leave it as is.  The clavrx products include product names that can get very general and then the AWIPS style rules might be applied incorrectly.  The other way to work around this would be to change the reader to alter the satpy name assigned to each datsaset_id?  Is that a better approach?  It seems a little odd.

Reply:
Nah, let's leave it.

Reply:
Ah I just realized, so this results in a physical_element global and variable attribute, right? I can live with that if so.

Reply:
Yes.  This is true.